src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2451,27 +2451,6 @@ internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
                     {
                         switch (declaredSymbol.Kind)
                         {
-                            case SymbolKind.Method:
-                                Debug.Assert((object)declaredSymbol.GetSymbol() == (object)ctor);
-                                return (node) =>
-                                       {
-                                           // Accept only nodes that either match, or above/below of a 'parameter list'/'base arguments list'.
-                                           if (node.Parent == recordDeclaration)
-                                           {
-                                               return node == recordDeclaration.ParameterList || node == recordDeclaration.BaseList;
-                                           }
-                                           else if (node.Parent is BaseListSyntax baseList)
-                                           {
-                                               return node == recordDeclaration.PrimaryConstructorBaseTypeIfClass;
-                                           }
-                                           else if (node.Parent is PrimaryConstructorBaseTypeSyntax baseType && baseType == recordDeclaration.PrimaryConstructorBaseTypeIfClass)
-                                           {
-                                               return node == baseType.ArgumentList;
-                                           }
-
-                                           return true;
-                                       };
```

Comment:
I think this is the root cause for duplication. If removing this will fail any other scenarios, we may try to find separate code paths for those different scenarios, or introduce a flag somewhere, etc..